### PR TITLE
Create `quantum_info.hamming_distance`

### PR DESCRIPTION
**Pull request by renatomello**
Checklist:
- [x] Reviewers confirm new code works as expected.
- [x] Tests are passing.
- [x] Coverage does not decrease.
- [x] Documentation is updated.

### CODE REVIEW

**Review by Simone-Bordoni — APPROVED** [review]

File: tests/test_quantum_info_utils.py [hunk 1]
```diff
@@ -45,6 +48,34 @@ def test_hamming_weight(bitstring, kind):
...
+    with pytest.raises(TypeError):
+        test = hamming_distance(2.3, "1010")
+    with pytest.raises(TypeError):
+        test = hamming_distance("1010", 2.3)
+
+    bitstring = f"{bitstring:b}"
+    distance = hamming_distance(kind(bitstring), kind(bitstring))
+    indexes = hamming_distance(kind(bitstring), kind(bitstring), True)
+
+    assert distance == 0
```
Comment (BrunoLiegiBastonLiegi) [inline tests/test_quantum_info_utils.py:75]: maybe add a non-zero test?

File: src/qibo/quantum_info/utils.py [hunk 2]
```diff
@@ -13,7 +13,9 @@
 from qibo.config import PRECISION_TOL, raise_error
 
 
-def hamming_weight(bitstring, return_indexes: bool = False):
+def hamming_weight(
+    bitstring: Union[int, str, list, tuple, np.ndarray], return_indexes: bool = False
+):
```
Comment (BrunoLiegiBastonLiegi) [inline src/qibo/quantum_info/utils.py:18]: definition of the weight?

File: tests/test_quantum_info_utils.py (hunk 1 above)
Comment (renatomello) [reply tests/test_quantum_info_utils.py]: Done

File: src/qibo/quantum_info/utils.py (hunk 2 above)
Comment (renatomello) [reply src/qibo/quantum_info/utils.py:18]: What do you mean?

Comment (BrunoLiegiBastonLiegi) [reply src/qibo/quantum_info/utils.py:18]: Sorry it's in the docstring, I didn't notice it before

**Review by BrunoLiegiBastonLiegi — APPROVED** [review]

**Pull request merged**